typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -295,12 +312,14 @@ export abstract class HyperlaneDeployer<
       'UpgradeBeacon',
       beaconDeployArgs,
     );
-    return this.deployProxy(
+    const contract = await this.deployProxy(
       chain,
       implementation as C,
       beacon.address,
       initArgs,
     );
+    this.cacheContract(chain, contractName, contract);
```

Comment:
why not `deployContractFromFactory` instead?

Reply:
I don't follow?